### DISCUSSION TITLE
Powerwall callibration

### DISCUSSION THREAD

**Discussion by NDGLB03** [netzero-labs/netzero#60] [Q&A] [answered]
I fell through the floor this morning when I found my Powerwall was charging even though it was 95% full, it was an expensive time to charge and it was set to discharge down to 25% according to the Netzero app. Running an automation did nothing to stop it charging.

Turning to the Tesla app supplied the answer, the system is running a callibration! It seems it may discharge below my minimum level (25%) as well and could take up to 24hours! 

I wish it would do it when power was cheap.

**Comment by BJReplay** [comment]
> Turning to the Tesla app supplied the answer, the system is running a callibration!

Fascinating! I have never seen this.

You didn't happen to capture any screen shots?

**Comment by NDGLB03** [comment]
![Screenshot_20241202_082918_Tesla](https://github.com/user-attachments/assets/7f3cd87d-b3c2-45e7-a8f4-4d26c3479908)
![Screenshot_20241202_083158_Tesla](https://github.com/user-attachments/assets/80041aab-67b1-484a-b71d-633ad5f8d9ab)
Your reply prompted me to take two. First time trying this so hopefully they will work.
I was too late to show it charging but you get the idea.

**Reply by BJReplay** [reply]
Thanks - can you by any chance click on the More Info link and see if that gets you to a web page that you can share?

**Comment by NDGLB03** [comment]
It doesn't. It's just a repeat of the same words.

**Comment by BJReplay** [comment]
How annoying, but thanks so much for sharing!

**Comment by zigam** [comment]
Powerwall 3 has a different battery chemistry (LFP vs NMC for Powerwall 2) which makes it harder to estimate the state of charge. LFP batteries need to be charged to 100% periodically to calibrate the state of charge (the same goes for cars with LFP batteries). The frequency and disruptive nature of calibration for Powerwall 3 is definitely an issue for some users.

**Reply by NDGLB03** [reply]
Mine is charged to 100% every night although sometimes the AI takes over if it thinks there's going to be lots of solar the next day.

**Comment by NDGLB03** [comment, marked as answer]
So, the callibration exercise appears to be completed. As it looked like the battery wanted to totally discharge itself, I plugged my car in late last night. IOG set up a charging plan and the car started charging. The battery ignored the Netzero automation and discharged itself, reaching fully discharged (Well, 1% according to Netzero) around 10:45pm. It then stayed discharged until 3am when it looks like it resumed normal service with time based charging. It only reached 85% charge before my cheap window ended but that's enough for me.

To sum up;
Yesterday was a bit of a shock but it only used a couple of kWh during the expensive rate. 
The battery still fed my house for the rest of the day until I plugged the car in.
Having the battery fully discharged left me vulnerable to not having blackout protection for a few hours.
I don't know if the battery needed to fully discharge or what would have happened had I not plugged the car in.

**Comment by NDGLB03** [comment]
UPDATE!It's calibrating again!

Today is a nice sunny day. Netzero looked odd, solar panel output being exported, battery discharging into the house demand. Bad timing as I've got a free electric slot early afternoon.

As I think about it, this could be a good arrangement so long as I don't run out of battery - export all my solar at 12 per KwH, import overnight at 7p per KwH.

**Comment by zigam** [comment]
Some of my observations regarding calibration:

## 1. Why is calibration needed?

Powerwall 3 uses a different battery chemistry compared to Powerwall 2: lithium iron phosphate (LFP) instead of nickel manganese cobalt (NMC). LFP batteries have some advantages (e.g. higher cycle life and lower cost) but also some disadvantages, including challenges with battery management system (BMS) calibration and cell balancing.

BMS calibration: Because LFP cells have a flatter voltage curve, estimating the state of charge is challenging, and requires periodic calibration. This is also why cars with LFP batteries recommend charging to 100% periodically.

Cell balancing: A battery pack is made up of many cells, and the voltage/state of charge across cells can become unbalanced. It's important to keep cells balanced, because it affects battery lifespan and safety.

## 2. Why does calibration take so long?

During calibration, Powerwall will charge up to 100%, then discharge to a low state of charge (which can go below backup reserve), then rest there a bit. Powerwall can use grid charging and discharging to speed up that process. However, this isn't always possible -- grid charging and exports might be turned off, or your grid import/export may be limited or disabled by the utility.

If grid export is not an option, you can accelerate the discharging process by adding house loads or charging an EV. If you have multiple batteries that need to discharge fully using only the baseline house load, the process can take a long time. That's likely why you see calibrations ranging anywhere from hours to days.

## 3. Why are calibrations so frequent?

The frequency varies from user to user. Some go months without calibration, while others seem to experience them every couple of weeks. The answer probably depends on the number of batteries, number of cycles, and depth of cycles.

Another question is whether Powerwall usage can be adjusted to make calibrations less frequent. Periodically charging the battery to 100% and doing a deep discharge would mimic a calibration process and might avoid one in the future. On the other hand, heavy use of the battery (lots of cycles with high charge/discharge rate) will probably contribute to cell imbalance.

**Comment by NDGLB03** [comment]
I'm not sure where I fit in the profile.

The battery gets charged to 100% overnight, if there is enough solar or if there is a free electric (Octopus power up) slot.
Outside of that, it rarely discharges below 40%.

It seems it will be calibrating every three months.
